### PR TITLE
[FIX] web: a field should always reach the model on 'change' event

### PR DESCRIPTION
**Pull request by SimonGenin**
Before this commit, a field would first check if the field was dirty
to reach the model for further validation and update.

However, if the following steps are followed:
- type a valid value (e.g. 10 in an integer field)
- make it invalid (e.g. 10*)
- make it valid again (e.g. 10)

The whole onChange code from the input fied hook would be bypassed as
the field would no be dirty anymore (because of the way it's handle
in the implementation). If it's bypassed, the field cannot know it's
no longer invalid. So the field would be underlined red and the record
would't be savable.

This commit being a fix in stable, we only make sure we always apply
the onChange logic of the hook by removing a condition.

opw 3680495

### CODE REVIEW

File: addons/web/static/tests/views/fields/integer_field_tests.js [hunk 1]
```diff
@@ -286,4 +287,31 @@ QUnit.module("Fields", (hooks) => {
             "Placeholder"
         );
     });
+
+    QUnit.test(
+        "a field is considered not dirty when the change event is triggered",
```
Comment (SimonGenin) [inline addons/web/static/tests/views/fields/integer_field_tests.js:292]: Not sure it's a great name

Comment (SimonGenin) [reply addons/web/static/tests/views/fields/integer_field_tests.js:292]: maybe "a field always validate it's value on change event"

File: addons/web/static/tests/views/fields/integer_field_tests.js [hunk 2]
```diff
@@ -286,4 +287,31 @@ QUnit.module("Fields", (hooks) => {
...
+            // The field was considered not dirty, so the onChange code wasn't executed, and the model still thought the value was invalid.
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "partner",
+                resId: 1,
+                arch: '<form><field name="int_field"/></form>',
+            });
+
+            const input = findElement(target, ".o_field_widget[name=int_field] input");
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/integer_field_tests.js:305]: I'm always careful with storing nodes in variables and interacting with them later on in the test. We can't guarantee that the node won't be replaced when the view is re-rendered, so I think it's safer to always querySelect.

Also, I'm not sure that findElement is necessary here. I would simply use querySelector

**Review by aab-odoo — APPROVED** [review]

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by FrancoisGe** [comment]
@robodoo r+

**Comment by SimonGenin** [comment]
@robodoo retry

**Comment by SimonGenin** [comment]
https://github.com/odoo/enterprise/pull/54983

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**